homeassistant/components/awattar/sensor.py

```diff
@@ -0,0 +1,225 @@
+"""Support for aWATTar sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+
+from homeassistant.components.sensor import (
+    DOMAIN as SENSOR_DOMAIN,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CURRENCY_EURO, UnitOfEnergy
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN, SERVICE_TYPE_DEVICE_NAMES
+from .coordinator import AwattarData, AwattarDataUpdateCoordinator
+
+
+@dataclass(frozen=True)
+class AwattarSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[AwattarData], float | datetime | None]
+    service_type: str
+
+
+@dataclass(frozen=True)
+class AwattarSensorEntityDescription(
+    SensorEntityDescription, AwattarSensorEntityDescriptionMixin
+):
+    """Describes a aWATTar sensor entity."""
+
+
+SENSORS: tuple[AwattarSensorEntityDescription, ...] = (
+    AwattarSensorEntityDescription(
+        key="current_hour_price",
+        translation_key="current_hour_price",
+        service_type="energy",
```

Comment:
Why not set as default in the dataclass above?

Reply:
removed `service_type' as it is not needed